boards/mega-xplained/doc.txt

```diff
@@ -10,27 +10,54 @@ is an evaluation kit by Atmel (now Microchip) for their ATmega1284P microcontrol
 
 ### Flash the board
 
-1. The board may be flashed through JTAG or using a SPI ISP programmer. If
-   the Buspirate is being used, then `make flash` can be used to flash the
-   board:
-```
-     make BOARD=mega-xplained -C examples/hello-world flash
-```
+You can change the fuses and flash the board using an SPI ISP programmer such
+as the Bus Pirate, or using a JTAG programmer such as the Atmel-ICE with a
+JTAG adapter.
+
+If using the Atmel-ICE as a JTAG programmer, connect the adapter in the
+orientation shown
+[here](https://gearfoxblog.files.wordpress.com/2016/12/img_20161221_172753.jpg?w=1000).
+
+1. First, make sure the default fuse settings are correct.<br/>
+
+   WARNING: setting the fuses incorrectly can brick your board!
 
-2. The default fuse settings must also be changed.<br/>
    If using the Buspirate:
+
 ```
      avrdude -p m1284p -c buspirate -P /dev/ttyUSB0 -U efuse:w:0xFF:m
      avrdude -p m1284p -c buspirate -P /dev/ttyUSB0 -U hfuse:w:0x99:m
      avrdude -p m1284p -c buspirate -P /dev/ttyUSB0 -U lfuse:w:0xE2:m
 ```
-WARNING: setting the fuses incorrectly can brick your board!
+
+   If using the Atmel-ICE:
+
+```
+     avrdude -p m1284p -c atmelice -U efuse:w:0xFF:m
+     avrdude -p m1284p -c atmelice -U hfuse:w:0x99:m
+     avrdude -p m1284p -c atmelice -U lfuse:w:0xE2:m
+```
+
+2. After the fuses are set, flash the application.
+   
+   If using the Buspirate:
+
+```
+     make BOARD=mega-xplained -C examples/hello-world flash
+```
+
+   If using the Atmel-ICE:
+
+```
+     make BOARD=mega-xplained PROGRAMMER=atmelice -C examples/hello-world flash
```

Comment:
The BOARD and PROGRAMMER should be put as environment variable, not a command line overwrite as it is not needed.

Reply:
With the current configuration, it does not work:

```
avrdude -p m1284p -c atmelice -P /dev/ttyUSB0 -F -U flash:w:/home/harter/work/git/RIOT/tests/bloom_bytes/bin/mega-xplained/tests_bloom_bytes.hex
avrdude: jtag3_open_common(): JTAGICE3/EDBG port names must start with "usb"

avrdude done.  Thank you.

/home/harter/work/git/RIOT/tests/bloom_bytes/../../Makefile.include:531: recipe for target 'flash' failed
make: *** [flash] Error 1
make: Leaving directory '/home/harter/work/git/RIOT/tests/bloom_bytes'
```

PROGRAMMER_FLAGS in the board `Makefile.include` should be put overwritable:

``` diff
diff --git a/boards/mega-xplained/Makefile.include b/boards/mega-xplained/Makefile.include
index 9a8d7e5f4..4d7e1b138 100644
--- a/boards/mega-xplained/Makefile.include
+++ b/boards/mega-xplained/Makefile.include
@@ -23,7 +23,7 @@ export DEBUGGER = $(DIST_PATH)/debug.sh $(DEBUGSERVER_FLAGS) $(DIST_PATH) $(DEBU
 # PROGRAMMER defaults to the Bus Pirate ISP
 export PROGRAMMER ?= buspirate
 
-export PROGRAMMER_FLAGS = -P /dev/ttyUSB0
+export PROGRAMMER_FLAGS ?= -P /dev/ttyUSB0
 
 OFLAGS += -j .text -j .data
 export FFLAGS += -p m1284p -c $(PROGRAMMER) $(PROGRAMMER_FLAGS) -F -U flash:w:$(HEXFILE)
```
and then the following works
```
BOARD=mega-xplained PROGRAMMER=atmelice PROGRAMMER_FLAGS= make -C tests/bloom_bytes/ flash term
```

Reply:
The other solution is to make the PROGRAMMER_FLAGS conditional to PROGRAMMER being `buspirate`.

Reply:
My bad, I was on another branch while testing this one. So no need to change PROGRAMMER_FLAGS.

Reply:
ok, so just to clarify: provide `BOARD` and `PROGRAMMER` via changing environment variables; no other changes?